test/e2e/common/pods.go

```diff
@@ -244,10 +244,6 @@ var _ = framework.KubeDescribe("Pods", func() {
 		pods, err := podClient.List(context.TODO(), options)
 		framework.ExpectNoError(err, "failed to query for pods")
 		framework.ExpectEqual(len(pods.Items), 0)
-		options = metav1.ListOptions{
-			LabelSelector:   selector.String(),
-			ResourceVersion: pods.ListMeta.ResourceVersion,
```

Comment:
this seems like a bug in the test... capturing the pre-create resource version here is intended to ensure the list/watch starts *before* the creation, but these options were not used properly in starting the informer. I think that means there's a race between the informer starting on line 273 and the pod create on line 277. I think this resourceVersion should maybe be used in the ListFunc on line 254; would be good to look at the history of this test and tag in the people involved in the most recent changes here (author/reviewer/approvers)

Reply:
I see that makes sense. 
I think the linter thinks this as ineffassign because we are trying to modify a field by a literal assignment.

Since the scope of options is same as the one in line 243, doing this made the linter happy.
`options.ResourceVersion = pods.ListMeta.ResourceVersion`

Reply:
I'd rather leave this as a failing check and open an issue for sig-node to dig into this test... catching unintentional bugs like this is exactly why we benefit from running static checks. In this case, I think it caught a bug, and I don't want to silence that signal by reworking the assignment to silence the check.

Reply:
Ack. Let me revert the changes on this file.